Source/It.cs

```diff
@@ -141,5 +141,35 @@ public static string IsRegex(string regex, RegexOptions options)
 			// But evaluated every time :)
 			return Match<string>.Create(value => re.IsMatch(value), () => It.IsRegex(regex, options));
 		}
+
+		/// <summary>
+		/// Allows for setting up open generic methods on Mock Fixtures
+		/// </summary>
+		public interface AnyType
+		{
+			/// <summary>
+			/// Will contain the specific object being compared inside an open generic method.
+			/// Used for checking inside It.Is&lt;It.AnyType&gt;(x => ((int)x.Object) == 2)
+			/// </summary>
+			object Object { get; set; }
+		}
+
+		internal class AnyTypeImplementation : AnyType
+		{
+			public object Object { get; set; }
+		}
```

Comment:
If I understand it correctly (guessing from your unit tests), the distinction between `interface AnyType` and `class AnyTypeImplementation` is needed so that one is able to bound / constrain type parameters. That would be a great feature to include. However I am not certain whether it will be obvious to most people how to use this feature. Do you think there would be a "more obvious" way of doing this, perhaps something along the lines of `It.AnySubTypeOf<TBase>` alongside `It.AnyType`?